.github/workflows/linter.yml

```diff
@@ -1,27 +1,50 @@
 name: Linter
 
 on:
-    push:
-        branches:
-            - main
-    pull_request:
-        branches:
-            - "*"
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "*"
 
 jobs:
-    solhint:
-        runs-on: ubuntu-20.04
-        steps:
-            - name: Checkout teleporter
-              uses: actions/checkout@v4
+  solhint:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout teleporter
+      uses: actions/checkout@v4
 
-            - name: Install solhint
-              run: |
-                  npm install solhint -g
-                  solhint --version
+    - name: Install solhint
+      run: |
+        npm install solhint -g
+        solhint --version
 
-            # "solhint **/*.sol" runs differently than "solhint '**/*.sol'", where the latter checks sol files
-            # in subdirectories. The former only checks sol files in the current directory and directories one level down.
-            - name: Run Lint
-              run: |
-                  ./scripts/lint.sh
+    # "solhint **/*.sol" runs differently than "solhint '**/*.sol'", where the latter checks sol files
+    # in subdirectories. The former only checks sol files in the current directory and directories one level down.
+    - name: Run Lint
+      run: |
+        ./scripts/lint.sh
+
+  golangci:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        path: teleporter
```

Comment:
nit: don't think we need to specify teleporter here

Reply:
This follows the pattern in the same job in awm-relayer: https://github.com/ava-labs/awm-relayer/blob/main/.github/workflows/linter.yml#L19

You're probably right, but let's keep this as is for consistency.